src/transaction/hooks/useSendCall.ts

```diff
@@ -0,0 +1,42 @@
+import type { Address } from 'viem';
+import { useSendTransaction as useSendCallWagmi } from 'wagmi';
+import { GENERIC_ERROR_MESSAGE } from '../constants';
+import type { UseSendCallParams } from '../types';
+import { isUserRejectedRequestError } from '../utils/isUserRejectedRequestError';
+
+/**
+ * Wagmi hook for single transactions with calldata.
+ * Supports both EOAs and Smart Wallets.
+ * Does not support transaction batching or paymasters.
+ */
+export function useSendCall({
+  setLifeCycleStatus,
+  transactionHashList,
+}: UseSendCallParams) {
+  const { status, sendTransactionAsync, data } = useSendCallWagmi({
```

Comment:
what are your thoughts on renaming this to be `sendCallAsync` instead of `sendTransactionAsync` for readability sake

Reply:
oh yeah, great point

Reply:
good call! added the alias